tokio/tests/io_util_empty.rs

```diff
@@ -0,0 +1,57 @@
+use std::time::Duration;
+use tokio::io::{empty, AsyncBufReadExt, AsyncReadExt};
+use tokio::time::timeout;
+
+#[tokio::test]
+async fn empty_read_is_cooperative() {
+    // the test would likely hang if reads on `empty` weren't
+    // cooperative
+    let _ = timeout(Duration::from_millis(1000), async {
+        loop {
+            let mut buf = [0u8; 4096];
+            let _ = tokio::io::empty().read(&mut buf).await;
+        }
+    })
+    .await;
+}
+
+#[tokio::test]
+async fn repeated_reads_on_same_empty_are_cooperative() {
+    // the test would likely hang if reads on `empty` weren't
+    // cooperative
+    let _ = timeout(Duration::from_millis(1000), async {
```

Comment:
These timeouts are too long when you aren't mocking the time driver. Please try reducing them to 5 ms or so.

Reply:
Honestly, selecting on the async block and `tokio::task::yield_now` would probably work too.